cmd/minishift/cmd/profile/profile_delete.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/machine/libmachine"
+	cmdUtil "github.com/minishift/minishift/cmd/minishift/cmd/util"
+	"github.com/minishift/minishift/pkg/minikube/cluster"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	profileActions "github.com/minishift/minishift/pkg/minishift/profile"
+	pkgUtil "github.com/minishift/minishift/pkg/util"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+var (
+	profileDeleteCmd = &cobra.Command{
+		Use:   "delete PROFILE_NAME",
+		Short: "delete profiles.",
+		Long:  "deletes an existing profile.",
+		Run:   runProfileDelete,
+	}
+	force bool
+)
+
+func runProfileDelete(cmd *cobra.Command, args []string) {
+	if len(args) == 0 {
+		atexit.ExitWithMessage(1, emptyProfileError)
```

Comment:
```
if len(args) == 0 || len(args) > 1 {
   atexit.ExitWithMessage(1, emptyProfileError)
```
would be good enough

Reply:
@gbraad I think we should have both the errors as it is a good thing to give proper error messages.

Reply:
If so, do not repeat the code. It should be in a function

Reply:
@gbraad I think checking arguments should be independent unless all sub commands do the same. It would have made more sense if all the sub commands i.e. list, set and delete need to have the same kind of arguments. But in this case it is just set and delete. So I feel it would be overkill. If you strongly feel then I can refactor it but I feel it is fine in current form.